transaction/jta/src/main/java/io/helidon/transaction/jta/JtaProvider.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.transaction.jta;
+
+import jakarta.transaction.TransactionManager;
+import jakarta.transaction.TransactionSynchronizationRegistry;
+import jakarta.transaction.UserTransaction;
+
+/**
+ * Jakarta Transaction service provider.
```

Comment:
What does this interface do? Why does it exist?

If I implement it like this:
```
public final class X implements JtaProvider {
  public final TransactionManager transactionManager() {
    return null;
  }
  public final UserTransaction userTransaction() {
    return null;
  }
  public final TransactionSynchronizationRegistry transactionSynchronizationRegistry() {
    return aRandomTsrEachTime();
  }
}
```
…what requirements of `JtaProvider` have I violated? That contractual information needs to exist here.